daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf_dev/daml_lf_1.proto

```diff
@@ -104,21 +112,22 @@ message ValName {
   // Module where the value is defined
   ModuleRef module = 1;
 
-  // *Must be a non-empty list of a valid identifiers unless `interned_id` is set.*
-  repeated string name = 2;
+  // *Must be a non-empty list of valid identifier*
+  repeated string name_dname = 2; // *Available in version < 1.dev*
 
-  // *Must be set if and only if `segments` is an empty list.*
-  int32 name_interned_id = 3; // *Available in versions >= 1.dev*
+  // *Must be a valid interned name*
+  int32 name_interned_dname = 3; // *Available in versions >= 1.dev*
 }
 
 // A field name definition in a record or a variant associated with a type.
 message FieldWithType {
 
-  // Name of the field .
-  // *Must be a valid identifier*
   oneof field {
-    string name = 1;
-    int32 interned_id = 3; // *Available in versions >= 1.dev*
+      // *Must be a valid identifier*
+      string field_str = 1; // *Available in version < 1.dev*
+
+      // *Must a valid interned identifier*
+      int32 field_interned_str = 3; // *Available in versions >= 1.dev*
```

Comment:
why `_str` suffix for interned field? `field_str` and `field_interned` (or interned_id) would feel more natural to me. (not a strong opinion). EDIT: I change my mind, having `interned_str` is good as we might have other types of interned things and it might not always be obvious what the type is.